subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/ExternalModuleDependencyVariantSpec.java

```diff
@@ -31,6 +31,12 @@
      */
     void platform();
 
+    /**
+     * Configures the dependency to select the "enforced-platform" variant.
+     * @since 7.3
+     */
+    void enforcedPlatform();
```

Comment:
Is there any way to not have this method appear here?
Ideally we expose `enforcedPlatform` in as little places as possible on the API.

Reply:
It's possible to avoid that if  `DependencyHandler.enforcedPlatform(Provider<MinimalExternalModuleDependency> dependencyProvider)` is implemented in DefaultDependencyHandler and not as default method (it needs to be added to DependencyHandlerDelegate then too).

From code point of view this is worse option. But if exposing enforcedPlatform is really that bad this is an option.

Another option is to add category method to ExternalModuleDependencyVariantSpec, but that seems even worse. 

Personally as a user I would expect enforcedPlatform to exist if there is a platform, since they seem related. What do you think?

Reply:
Discussed this further and give the history of `enforcedPlatform`, let's not add it to `ExternalModuleDependencyVariantSpec` for now